### ISSUE TITLE
No template found for residue 645 (G)

### ISSUE THREAD

**Issue by leosv123** [openmm/openmm#4512]
ValueError: No template found for residue 645 (G).  The set of atoms matches G3, but the bonds are different.  Perhaps the chain is missing a terminal group?  For more information, see https://github.com/openmm/openmm/wiki/Frequently-Asked-Questions#template

May I know how to resolve this issue why is it occurring? 

Any help please

[test_cif_fixed_pH_7.pdb.zip](https://github.com/openmm/openmm/files/14972021/test_cif_fixed_pH_7.pdb.zip)

**Comment by peastman** [comment]
Did you follow the link in the error message and read the explanation?  It describes the possible causes pretty well.

**Comment by leosv123** [comment]
Yeah I check throught the link and tried fixing some issues using PDBFixer, changed some residue names too at last ended up with this issue trying to figure it out. This works good in gromacs using AMBER99SB and AMBER94 forcefield 

here I am using

`
from openmm.app import *
from openmm import *
from openmm.unit import *
from sys import stdout

pdb = PDBFile('/home/lingrajsvannur/openmm/clean/test_cif_fixed_pH_7.pdb')

forcefield = ForceField('amber14-all.xml', 'amber14/tip3pfb.xml')
system = forcefield.createSystem(pdb.topology, nonbondedMethod=PME,
        nonbondedCutoff=1*nanometer, constraints=HBonds)`

**Comment by leosv123** [comment]
As I am still facing issue in loading Pdb in OpenMM directly


And since My pdb is loading perfectly in gromacs, I used the gro and top files as input to the Openmm system, it works 
but I want to do some modifications like remove some water molecules and then run the MD simulation. But after the removal of water molecules I am not able to save the updated system


from openmm.app import GromacsTopFile, GromacsGroFile, PDBFile, Simulation
from openmm import LangevinMiddleIntegrator, VerletIntegrator
import openmm.unit as unit
import openmm as mm
from openmm.app import *
from openmm import *
from openmm.unit import *
from sys import stdout

# Load the GROMACS files
gro_file = 'solvated.gro' 
top_file = 'topol.top'    

gro = GromacsGroFile(gro_file)
top = GromacsTopFile(top_file, periodicBoxVectors=gro.getPeriodicBoxVectors()) 

# Create the system
system = top.createSystem(nonbondedMethod=PME, nonbondedCutoff=1.0*nanometer, constraints=HBonds)
modeller = Modeller(top.topology, gro.positions)


make some modifications and delete few water moelcules 

modeller.delete(list(water_to_delete))


Now how to use the updated modeller to run the simulation as I need to build a updated system to run simulation correct? But I am not able to save in gro or top format to and create updated system. This is raising problem too.

**Comment by peastman** [comment]
The chain is terminated incorrectly.  The Amber force field expects the phosphate to be missing at the 5' end, replaced with just a hydrogen on O5'.  Your file includes both the phosphate and the hydrogen, meaning that you have O5' bonded to three other atoms.
